gateway/gateway.go

```diff
@@ -86,9 +86,20 @@ func NewGateway() Gateway {
 	}
 }
 
-// returns a copy of the given url, with the host potentially resolved using a
-// srv request
-func (g Gateway) resolveURL(uu *url.URL) *url.URL {
+// NewGatewayWithSRVClient returns an instantiated Gateway object
+// It accepts a SRVClient if you have a custom instance
+func NewGatewayWithSRVClient(srv *srvclient.SRVClient) Gateway {
```

Comment:
Instead of making a new method I think we should make the srvclient field public on the Gateway struct, so we don't have to copy-paste a bunch of code